tests/foreman/cli/test_discoveredhost.py

```diff
@@ -14,50 +14,135 @@
-    """Check if host is discovered and information about it can be
-    retrieved back
+@pytest.mark.e2e
+@pytest.mark.on_premises_provisioning
+@pytest.mark.parametrize('module_provisioning_sat', ['discovery'], indirect=True)
+@pytest.mark.parametrize('pxe_loader', ['bios', 'uefi'], indirect=True)
+@pytest.mark.rhel_ver_match('8')
+def test_positive_provision_pxe_host_cli(
+    module_provisioning_rhel_content,
+    module_discovery_sat,
+    provisioning_host,
+    provisioning_hostgroup,
+):
+    """Provision a pxe-based discovered host from CLI
 
-    Introduced a delay of 300secs by polling every 10 secs to get expected
-    host
-    """
-    for _ in range(30):
-        try:
-            discovered_host = DiscoveredHost.info({'name': hostname})
-        except CLIReturnCodeError:
-            sleep(10)
-            continue
-        return discovered_host
+    :id: 91bb254b-3c30-4608-b1ba-e18bcc22efb5
 
+    :parametrized: yes
 
-@pytest.mark.stubbed
-@pytest.mark.tier3
-def test_positive_pxe_based_discovery():
-    """Discover a host via PXE boot by setting "proxy.type=proxy" in
-    PXE default
+    :Setup: Provisioning and discovery should be configured
 
-    :id: 25e935fe-18f4-477e-b791-7ea5a395b4f6
-
-    :Setup: Provisioning should be configured
+    :Steps:
 
-    :Steps: PXE boot a host/VM
+        1. Boot up the host to discover
+        2. Provision the host
 
-    :expectedresults: Host should be successfully discovered
+    :expectedresults: Host should be provisioned successfully from CLI
 
     :CaseImportance: Critical
 
     :BZ: 1731112
     """
+    sat = module_discovery_sat.sat
+    provisioning_host.power_control(ensure=False)
+    mac = provisioning_host._broker_args['provisioning_nic_mac_addr']
+
+    wait_for(
+        lambda: sat.api.DiscoveredHost().search(query={'mac': mac}) != [],
+        timeout=240,
+        delay=20,
+    )
+    discovered_host = sat.api.DiscoveredHost().search(query={'mac': mac})[0]
+    discovered_host.hostgroup = provisioning_hostgroup
+    discovered_host.location = provisioning_hostgroup.location[0]
+    discovered_host.organization = provisioning_hostgroup.organization[0]
+    discovered_host.build = True
+    result = sat.cli.DiscoveredHost.provision(
+        {
+            'id': discovered_host.id,
+            'hostgroup-id': discovered_host.hostgroup.id,
+            'organization-id': discovered_host.organization.id,
+            'location-id': discovered_host.location.id,
+        }
+    )
+    assert 'Host created' in result[0]['message']
+    host = sat.api.Host().search(query={"search": f'id={discovered_host.id}'})[0]
+    assert host
+    wait_for(
+        lambda: host.read().build_status_label != 'Pending installation',
+        timeout=1500,
+        delay=10,
+    )
+    host = host.read()
+    assert host.build_status_label == 'Installed'
+    host.delete()
+    assert not sat.api.Host().search(query={"search": f'id={discovered_host.id}'})
```

Comment:
if host fails to provision with Installed state, then assert could fail and host will exist in sat, so can we cover this in request.addfinalizer?

Reply:
Good catch.